db/db_compaction_test.cc

```diff
@@ -74,6 +74,47 @@ class CompactionStatsCollector : public EventListener {
   std::vector<std::atomic<int>> compaction_completed_;
 };
 
+class DeletionTriggeredCompactionWithMinFileSizeTestListener
+    : public EventListener {
+ public:
+  void OnCompactionBegin(rocksdb::DB* db,
+                         const rocksdb::CompactionJobInfo& ci) override {
+    if (ci.compaction_reason != CompactionReason::kFilesMarkedForCompaction) {
+      std::cerr << "Actual compaction reason is "
+                << static_cast<int>(ci.compaction_reason)
+                << ": expected: kFilesMarkedForCompaction" << std::endl;
+      return;
+    }
```

Comment:
Is it possible to guarantee only compaction of CompactionReason::kFilesMarkedForCompaction will happen in your test so that you can assert `ci.compaction_reason == CompactionReason::kFilesMarkedForCompaction`. This is a more common practice to write consistent and effective test.

Reply:
> This is a more common practice to write consistent and effective test.

I don't know if our testing infra monitor std::cerr and tell us there is an unexpected behavior like assertion failure does. Your check later about file count may correctly fail because of this but it's subtle to figure out.

Reply:
Yes - assertion is enough and it will crash on error. Just to make sure the test is set up in a way that only deletion triggered compaction is triggered. If you look at `LevelCompactionBuilder::PickCompaction()`, it tries to pick files for other compaction before deletion triggered ones (`PickFilesMarkedForCompaction()`). We don't want other compaction to get picked so the assertion won't fail. 

Usually you can achieve that by setting a high level0_file_num_compaction_trigger and a high max_bytes_for_level_base.